ibm/service/kms/resource_ibm_kms_key.go

```diff
@@ -415,3 +336,66 @@ func KmsEndpointURL(kpAPI *kp.Client, endpointType string, extensions map[string
 	}
 	return u, nil
 }
+
+// Extract and Validate data from schema related to a key
+func ExtractAndValidateKeyDataFromSchema(d *schema.ResourceData, meta interface{}) (kpAPI *kp.Client, key kp.Key, err error) {
+	instanceID := getInstanceIDFromCRN(d.Get("instance_id").(string))
+	kpAPI, _, err = populateKPClient(d, meta, instanceID)
```

Comment:
populateKPClient() has a different functionality than extracting and formatting the data. So, this can still be the responsibility of the calling function. The calling function can do this, at high level:
1. ExtractAndValidateKeyDataFromSchema()
2. populateKPClient()
3. Invoke CreateKey or CreateKeyWithPolicies()
4. Response process